### ISSUE TITLE
LayerIntegratedGradients for loop model

### ISSUE THREAD

**Issue by Ryosuke0104** [pytorch/captum#1050]
## ❓ Questions and Help

I'm newcomer of Captum, and thank you for your amazing work.

I'm trying to interpret my model by using LayerIntegratedGradients.
My model's summary is
`class Mymodel(nn.Module):
   def __init__():
        .
        .
        .
   def forward(self, qa_list):

       batch_size = 64
       foo = 31
       bar = 15
       word_embedding_dim = 512
       
       # qa_list.size() : [batch_size, foo, bar]

       text_enc_hidden = torch.tensor(torch.zeros([batch_size, foo, embedding_dim]]))
        for i in range(foo)):

            history = qa_list[:, i] # [batch_size, foo]

            history_embedding = self.word_embeddings(history) 
            _hidden = self.lstm(history_embedding

            text_enc_hidden[:, i, :] = _hidden.squeeze(1)

       return text_enc_hidden
`
What I want to say is "I use for loop in forward function"
LayerIntegratedGradients can be used for this model?

I compute LayerIntegratedGradients like below

`  
  # compute attributions and approximation delta using layer integrated gradients
  attributions_ig, delta = lig.attribute(inputs=(input_indices, input_dialog_len), 
                                         baselines=(baseline_input_indices, baseline_input_dialog_len),
                                         target=pred_ind,
                                         n_steps=500, 
`
But, attributions_ig.size() is different from input_indices.size()

input_indices.size() : [1, 300]
attributions_ig.size(): [1, 31, 512]  = [1, foo, word_embedding_dim])

I wonder whether LayerIntegratedGradients can be used for Model which include for loop

Thank you for your kindness.

**Comment by craymichael** [comment]
Hi, this is covered by the LayerIntegratedGradients API behavior. Layer attribution returns attributions with the selected layer input/output shape, not necessarily the original model input shape; loops in `forward` are fine as long as the selected layer and scalar target are well defined. Closing as documented. https://captum.ai/api/layer.html#layer-integrated-gradients
